plugins/backup/veeam/src/main/java/org/apache/cloudstack/backup/veeam/VeeamClient.java

```diff
@@ -324,6 +316,21 @@ private boolean checkTaskStatus(final HttpResponse response) throws IOException
         return false;
     }
 
+    protected boolean checkIfRestoreSessionFinished(String type, String path) throws IOException {
+        for (int j = 0; j < this.restoreTimeout; j++) {
+            HttpResponse relatedResponse = get(path);
+            RestoreSession session = parseRestoreSessionResponse(relatedResponse);
+            if (session.getResult().equals("Success")) {
+                return true;
+            }
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException ignored) {
+            }
```

Comment:
I think it would be good, at least, to put a trace log here.

Reply:
Done, thanks for the suggestion